app/models/piece.rb

```diff
@@ -64,7 +64,11 @@ def is_obstructed?(row_final, col_final)
 
       #diagonal case
     elsif ((row_final - row_init).to_f/(col_final - col_init).to_f ).abs == 1
-      return Piece.exists?(position_row: row_range, position_column: col_range, game_id: game_id)
+      i = 0
```

Comment:
Hey @Edwin-F  So the problem was that the logic was testing too many possibilities.
Basically, for the row range you would get [x,y,z] and for the col range you would want [q,r,s].
The postions you want tested are (x,q), (y,r) and (z,s).

However, the way it was written before was testing all combinations.
It tested (x,q), (x,r) (x,s)...etc.

This caused the board to say the bishop was obstructed even when it was not because it was testing too many locations.

I didn't write any new tests, thats why your old tests are still passing.